### PR TITLE
Feature bluetooth v2

### PR DESCRIPTION
**Pull request by vimian**

### CODE REVIEW

File: src/main/main.c [hunk 1]
```diff
@@ -163,6 +171,131 @@ void listen_to_buttons (void *pvParameters) {
     }
 }
 
+esp_bd_addr_t trusted_device;
+bool paired = false;
+
+static char *bda2str(esp_bd_addr_t bda, char *str, size_t size) {
```
Comment (SSpedsbjerg) [inline src/main/main.c:177]: ikke tydeligt hvad den gør, eller hvorfor den returnere en reference til en karakter

File: src/main/main.c [hunk 2]
```diff
@@ -163,6 +171,131 @@ void listen_to_buttons (void *pvParameters) {
...
+    if (bda == NULL || str == NULL || size < 18) {
+        return NULL;
```
Comment (SSpedsbjerg) [inline src/main/main.c:179]: Vil bare lige advare at NULL ikke eksistere i C/C++, du vil returnere og eller tjekke for tallet 0. vær sikker på at det ikke er et scenarie som give problemer

File: src/main/main.c [hunk 3]
```diff
@@ -163,6 +171,131 @@ void listen_to_buttons (void *pvParameters) {
...
+
+            if (is_locked && !unlocking) {
+                unlocking = true;
+                unlock_timer = 0;
+            }
+            break;
+        }
+        case ESP_BT_GAP_ACL_DISCONN_CMPL_STAT_EVT: {
+            //printf("Device disconnected status: %d\n", param->acl_disconn_cmpl_stat.reason);
+            paired = true;
```
Comment (SSpedsbjerg) [inline src/main/main.c]: Jeg ser ikke grunden til at du smider paired i true her.

File: src/main/main.c (hunk 1 above)
Comment (vimian) [reply src/main/main.c:177]: It is a default function from ESP IDF, to convert a bda to a MAC like string.

File: src/main/main.c (hunk 2 above)
Comment (vimian) [reply src/main/main.c:179]: Same as before.

File: src/main/main.c (hunk 3 above)
Comment (vimian) [reply src/main/main.c]: Yeah, I see the issue, I will fix it.

Comment (vimian) [reply src/main/main.c]: Changed to only set paired = true, on pairing completion, instead of every time a device disconnects.

Comment (vimian) [reply src/main/main.c]: Never mind, I changed it back to support all cases instead. It is better described in the comments I added to the code.

Comment (SSpedsbjerg) [reply src/main/main.c]: I don't think that you have clarified it, could you further describe the reasoning

**Review by GachiRoss — DISMISSED** [review]

File: src/main/main.c [hunk 4]
```diff
@@ -163,6 +171,135 @@ void listen_to_buttons (void *pvParameters) {
...
+                unlock_timer = 0;
+            }
+            break;
+        }
+        case ESP_BT_GAP_ACL_DISCONN_CMPL_STAT_EVT: {
+            //printf("Device disconnected status: %d\n", param->acl_disconn_cmpl_stat.reason);
+            /*if (param->acl_disconn_cmpl_stat.reason == ESP_BT_STATUS_HCI_PEER_USER) {
+                paired = true;
+            }*/ // Works only when device gets to call the disconnect after pairing
+            paired = true; // Works always
```
Comment (SSpedsbjerg) [inline src/main/main.c:234]: "To ensure the finishing of the handshake between the phone and ESP, the paired has to be true"

**Review by SSpedsbjerg — APPROVED** [review]

**Review by GachiRoss — APPROVED** [review]
seeemsGood

**Pull request merged**